### PR TITLE
Fix #10032 - Update strings/styles of VPN platform pages 

### LINKED ISSUE #10032: Update VPN platform pages to match VPN landing page
## Description

The VPN landing page was updated on Mon, March 22. The VPN platform pages need to be revised to match the new layout and migrate strings into the new shared FTL file. 

- [ ] /products/vpn/desktop/    
- [ ] /products/vpn/desktop/linux/
- [ ] /products/vpn/desktop/mac/
- [ ] /products/vpn/desktop/windows/
- [ ] /products/vpn/mobile/
- [ ] /products/vpn/mobile/android/
- [ ] /products/vpn/mobile/ios/

Part of Epic #9963

---

#### Success Criteria
- [ ] Each page should match the design styles of the homepage
- [ ] Each platform page should use the new shared FTL file 
- [ ] Approved by stakeholders

### PR DESCRIPTION
**Pull request by maxxcrawford**
## Description

Update the design and FTL strings to match the VPN landing page

TODO:

- [x] Remove [`special-templates.toml`](https://github.com/mozilla/bedrock/blob/10032-vpn-platform-page-updates/l10n/configs/special-templates.toml) rule to keep the platform pages out of Pontoon. (Pending approval from @peiying2)
- [ ] Determine (and then act on) how to handle the similar strings in [`shared.ftl`](https://github.com/mozilla/bedrock/blob/10032-vpn-platform-page-updates/l10n/en/products/vpn/shared.ftl)

## Issue / Bugzilla link

#10032 

## Testing

-  http://localhost:8000/en-US/products/vpn/desktop/    
-  http://localhost:8000/en-US/products/vpn/desktop/linux/
-  http://localhost:8000/en-US/products/vpn/desktop/mac/
-  http://localhost:8000/en-US/products/vpn/desktop/windows/
-  http://localhost:8000/en-US/products/vpn/mobile/
-  http://localhost:8000/en-US/products/vpn/mobile/android/
-  http://localhost:8000/en-US/products/vpn/mobile/ios/

### CODE REVIEW

File: bedrock/products/templates/products/vpn/includes/macros.html
Comment (maxxcrawford) [inline bedrock/products/templates/products/vpn/includes/macros.html:57]: Created a new marco that can define a single image (which vary between desktop and mobile platform pages)

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [inline bedrock/products/templates/products/vpn/platforms/desktop.html:5]: TODO: Remove `vpn_platform_content_well` macro

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [inline bedrock/products/templates/products/vpn/platforms/desktop.html:25]: This PR will remove the need for the platform-specific CSS bundle. Does it make sense to use `-landing`, or should I re-fork the styles into the `platforms` subfolder?

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [inline bedrock/products/templates/products/vpn/platforms/desktop.html:60]: Updated page to use the `vpn_platform_content_well` macro.

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [inline bedrock/products/templates/products/vpn/platforms/desktop.html:134]: Using the `vpn_content_block` call as seen in the `pricing-fixed.html` include. I added a modifer class to center the elements (per JR's direction).

File: l10n/en/products/vpn/shared.ftl
Comment (maxxcrawford) [inline l10n/en/products/vpn/shared.ftl:28]: Ported over items from the [`products/vpn/platforms/shared.ftl`](https://github.com/mozilla/bedrock/blob/10032-vpn-platform-page-updates/l10n/en/products/vpn/platforms/shared.ftl) file that did not exist in the root shared FTL file.

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/block.scss:77]: If there's a better naming convention, please advise!

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [inline media/css/products/vpn/components/block.scss:102]: I noticed that this file had the media queries below, rather than nested. LMK if I need to move this down accordingly.

**Review by maxxcrawford — COMMENTED** [review]
Self-review notes for the desktop page. After approval of this approach, I will port the rest over (also in this PR) 

Note that for the strings, I started a [spread sheet](https://docs.google.com/spreadsheets/d/1Z4XvNBpoCrzBu-KJcdZHgFjah7kV03HtqR_hBjjwbCY/edit?usp=sharing) for easier find-replacing.

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (alexgibson) [reply bedrock/products/templates/products/vpn/platforms/desktop.html:25]: I tried to make things modular so you could compile your own bundle, as opposed to rely on the stylesheet for another page. But if there are things in landing.scss that you need, then we should make those shared components.

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (alexgibson) [inline bedrock/products/templates/products/vpn/platforms/desktop.html:52]: Try not to rely on conditional classes such as `js-text-vpn-fixed-price-available`. Those are undergoing quite a big refactor to support the new pricing models, so I would try and keep code related to this logic out of these pages for now.

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (alexgibson) [reply bedrock/products/templates/products/vpn/platforms/desktop.html:134]: Is there a more generic name we could use for this modifier class, something that's not associated with the pages where it is used?

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [reply l10n/en/products/vpn/shared.ftl:28]: This is very simular to the string above `No logging of your network activity`. Rather than adding more translations for essentially the same sentance, I'd suggest talking with Raphael to see if there are strings we could standardize here across pages?

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [inline l10n/en/products/vpn/shared.ftl:32]: Same as above, this is a very similar sentance to what we have translated already.

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [inline l10n/en/products/vpn/shared.ftl:40]: Same as above, this is a very similar sentance to what we have translated already.

File: media/css/products/vpn/components/block.scss
Comment (alexgibson) [reply media/css/products/vpn/components/block.scss:77]: I'd say let's use something that describes the layout that the modifier creates, rather than something that ties it to the platform pages?

**Review by alexgibson — CHANGES_REQUESTED** [review]
I haven't pulled this branch yet but I took a look and left some code-level comments. Note we'll have to also undo the L10n config that currently excludes the platform strings from being exposed in Pontoon.

**Comment by maxxcrawford** [comment]
> I haven't pulled this branch yet but I took a look and left some code-level comments. Note we'll have to also undo the L10n config that currently excludes the platform strings from being exposed in Pontoon.

Awesome. Thank you!

File: l10n/en/products/vpn/shared.ftl
Comment (maxxcrawford) [reply l10n/en/products/vpn/shared.ftl:32]: Yeah. I think part of the challenge is trying to having different verbage for SEO purposes.

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [reply bedrock/products/templates/products/vpn/platforms/desktop.html:134]: I think so – it's basically just `t-center`.

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [reply bedrock/products/templates/products/vpn/platforms/desktop.html:52]: `js-text-vpn-fixed-price-available` was copy/paste artifacting. I'll remove it.

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [reply bedrock/products/templates/products/vpn/platforms/desktop.html]: Copy that. I went back and removed landing from the bundle. It was not needed!

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [reply bedrock/products/templates/products/vpn/platforms/desktop.html]: Renamed to `.t-center`

File: l10n/en/products/vpn/shared.ftl
Comment (maxxcrawford) [reply l10n/en/products/vpn/shared.ftl:28]: @alexgibson Who do we need to sync up with for this? I'm happy to make the switch if there's no SEO-related objections.

File: bedrock/products/templates/products/vpn/platforms/desktop.html
Comment (maxxcrawford) [reply bedrock/products/templates/products/vpn/platforms/desktop.html]: Removed!

**Comment by maxxcrawford** [comment]
Marking this PR as ready to go. There are to small outstanding TODOs that need to be addressed before merging, but they aren't blocking getting the review underway. 🤝

File: l10n/en/products/vpn/shared.ftl
Comment (alexgibson) [reply l10n/en/products/vpn/shared.ftl:28]: I'd suggest talking to Raphael on this. If there are obvious keywords that add value then it makes sense, but for some of these strings they are already very similar, and we should avoid creating more work for localizers than necessary. I'll defer to what you ultimately decide here.

**Review by alexgibson — CHANGES_REQUESTED** [review]
A few comments / suggestions, but mostly nits and certainly no blockers.

It's a bit unfortunate the images are sized in such a way that the components need the theme class workaround. But if resizing the images now is going to be a pain then the modifier works OK for now 👍

File: bedrock/products/templates/products/vpn/platforms/android.html
Comment (alexgibson) [inline bedrock/products/templates/products/vpn/platforms/android.html:83]: Ooh I forgot to call this out. We should remove "Refund Policy" here as it's not really a product feature, and should only be accompanied underneath a subscribe button. We're making the same change on the main landing page to reflect this.

File: bedrock/products/templates/products/vpn/includes/macros.html
Comment (maxxcrawford) [reply bedrock/products/templates/products/vpn/includes/macros.html:69]: Removed!

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [reply media/css/products/vpn/components/block.scss:83]: 🙈 Less _is_ more.

File: bedrock/products/templates/products/vpn/platforms/android.html
Comment (maxxcrawford) [reply bedrock/products/templates/products/vpn/platforms/android.html:83]: Good call out. Removed!

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [reply media/css/products/vpn/components/block.scss:94]: Copy that. Confirmed and removed.

File: media/css/products/vpn/components/block.scss
Comment (maxxcrawford) [reply media/css/products/vpn/components/block.scss]: I removed this class from SCSS and the param in the macro function. I revised some of the baseline styles to mirror this better.

**Review by alexgibson — APPROVED** [review]
🔍 r+

**Pull request merged**
